src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +18,45 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't create phone: " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            Predicate finalPredicate = criteriaBuilder.and();
+            for (Map.Entry<String, String[]> entrySet : params.entrySet()) {
+                CriteriaBuilder.In<String> currentPredicate =
+                        criteriaBuilder.in(root.get(entrySet.getKey()));
+                for (String value : entrySet.getValue()) {
+                    currentPredicate.value(value);
+                }
+                finalPredicate = criteriaBuilder.and(finalPredicate, currentPredicate);
+            }
+            query.where(finalPredicate);
+            return session.createQuery(query).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find phones with params: " + params, e);
```

Comment:
the output of toString() called on  "params"  will give you something like this:
![image](https://user-images.githubusercontent.com/22300600/128191429-8e0f3a91-ab10-46ec-9a0f-b38b67b107e0.png)